### PR TITLE
API doc

### PR DESCRIPTION
**Pull request by stigrj**
Cleaning up the API doc to make it more formal

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#!/usr/bin/env python
```
Comment (robertodr) [inline doc/conf.py]: You should have a look at https://github.com/dftlibs/xcfun/blob/master/docs/conf.py I simplified it a lot with respect to this version here.

**Comment by stigrj** [comment]
This is now ready, and with this I think we have a new release v1.2

File: doc/mrcpp_api/introduction.rst [hunk 2]
```diff
@@ -74,7 +86,7 @@ which can be written as the lambda function
 
 .. code-block:: cpp
 
-    auto Z = 1.0;                                 // Hydrogen nuclear charge
+    auto Z = 1.0; // Hydrogen nuclear charge
     auto f = [Z] (const mrcpp::Coord<3> &r) -> double {
```
Comment (bjorgve) [inline doc/mrcpp_api/introduction.rst]: We can kill the `-> double`. I tested it in pilot

File: doc/mrcpp_api/mwoperators.rst [hunk 3]
```diff
@@ -192,8 +184,9 @@ BSOperator
 ++++++++++
 
 The BS derivative operator is based on a pre-projection onto B-splines in order
-to remove the discontinuities in the MW basis. This operator is also available
-as a direct second and third derivative.
+to remove the discontinuities in the MW basis, see `Anderson etal.
```
Comment (bjorgve) [inline doc/mrcpp_api/mwoperators.rst]: `etal. -> et al.`  ?

File: doc/mrcpp_api/mwoperators.rst [hunk 4]
```diff
@@ -144,7 +136,7 @@ ABGVOperator
 The ABGV (Alpert, Beylkin, Gines, Vozovoi) derivative operator is initialized
 with two parameters :math:`a` and :math:`b` accounting for the boundary
 conditions between adjacent nodes, see `Alpert etal.
```
Comment (bjorgve) [inline doc/mrcpp_api/mwoperators.rst]: `etal. -> et al.`  ?

File: src/treebuilders/add.cpp [hunk 5]
```diff
@@ -76,26 +76,26 @@ void add(double prec,
...
 }
 
-/** @brief Addition of several MW function representations
+/** @brief Addition of several MW function representations, adaptive grid
  *
- * @param[in] prec Build precision of output function
- * @param[in,out] out Output function to be built
+ * @param[in] prec: Build precision of output function
+ * @param[out] out: Output function to be built
  * @param[in] inp Vector of input function
```
Comment (bjorgve) [inline src/treebuilders/add.cpp]: missing colon?

File: src/treebuilders/grid.cpp [hunk 6]
```diff
@@ -250,12 +274,13 @@ template <int D> int refine_grid(FunctionTree<D> &out, double prec, bool absPrec
 /** @brief Refine the grid of a MW function representation
  *
  * @param[in,out] out Output tree to be refined
- * @param[in] inp Input tree
+ * @param[in] inp Input tree that defines the new grid
```
Comment (bjorgve) [inline src/treebuilders/grid.cpp]: missing colon(s) after `inp` and `out`.

File: src/utils/Plotter.cpp [hunk 7]
```diff
@@ -47,28 +46,49 @@ Plotter<D>::Plotter(const Coord<D> &o)
...
-*/
+/** @brief Grid plot of a MWTree
+ *
+ *  @param[in] tree: MWTree to plot
+ *  @param[in] fname: File name for output, without extension
+ *
+ *  @details Writes a file named fname + file extension (".grid" as default)
+ *  to be read by geomview to visualize the grid (of endNodes) where the
+ *  multiresolution function is defined. In MPI, each process will write a
+ *  separate file, and will print only nodes owned by itself (pluss the
```
Comment (bjorgve) [inline src/utils/Plotter.cpp]: and will only print

File: src/utils/Printer.h [hunk 8]
```diff
@@ -41,20 +41,68 @@ namespace mrcpp {
...
     static void setScientific() { *out << std::scientific; }
+
+    /** @brief Use fixed floating point notation, e.g. 0.01 */
     static void setFixed() { *out << std::fixed; }
 
-    static int setWidth(int i);
-    static int setPrecision(int i);
-    static int setPrintLevel(int i);
+    /** @brief Set new line width for printed output
+     *  @param[in] i New width (number of characters)
```
Comment (bjorgve) [inline src/utils/Printer.h]: colon after i

File: src/utils/Printer.h [hunk 9]
```diff
@@ -41,20 +41,68 @@ namespace mrcpp {
...
+     *  @returns Old width (number of characters)
+     */
+    static int setWidth(int i) {
+        int oldWidth = printWidth;
+        printWidth = i;
+        return oldWidth;
+    }
+
+    /** @brief Set new precision for floating point output
+     *  @param[in] i New precision (digits after comma)
```
Comment (bjorgve) [inline src/utils/Printer.h]: colon after i

File: src/utils/Printer.h [hunk 10]
```diff
@@ -41,20 +41,68 @@ namespace mrcpp {
...
+     */
+    static int setPrecision(int i) {
+        int oldPrec = printPrec;
+        printPrec = i;
+        *out << std::setprecision(i);
+        return oldPrec;
+    }
 
+    /** @brief Set new print level
+     *  @param[in] i New print level
```
Comment (bjorgve) [inline src/utils/Printer.h]: ditto

File: doc/mrcpp_api/parallel.rst [hunk 11]
```diff
@@ -62,19 +62,16 @@ the shared ``FunctionTree``.
...
 
+.. doxygenfunction:: mrcpp::send_tree(FunctionTree<D>&, int, int, MPI_Comm, int)
+.. doxygenfunction:: mrcpp::recv_tree(FunctionTree<D>&, int, int, MPI_Comm, int)
+
+Example
++++++++
+
 A blocking send/receive means that the function call does not return until the
 communication is completed. This is a simple and safe option, but can lead to
 significant overhead if the communicating MPI prosesses are not synchronized.
```
Comment (robertodr) [inline doc/mrcpp_api/parallel.rst]: Small typo "prosesses" --> "processes"

File: doc/mrcpp_api/parallel.rst [hunk 12]
```diff
@@ -110,6 +107,17 @@ recv_tree
...
+        MRCPP
+    :members:
+        SharedMemory
+
+.. doxygenfunction:: mrcpp::share_tree(FunctionTree<D>&, int, int, MPI_Comm)
+
+Example
++++++++
+
 The sharing of a ``FunctionTree`` happends in three steps: first a
```
Comment (robertodr) [inline doc/mrcpp_api/parallel.rst]: Typo "happends" --> "happens"

**Comment by stigrj** [comment]
Looks like the `@param` gets a colon in the docs regardless

File: doc/mrcpp_api/introduction.rst (hunk 2 above)
Comment (stigrj) [reply doc/mrcpp_api/introduction.rst]: It's still expressive in the context of documentation, though

Comment (robertodr) [reply doc/mrcpp_api/introduction.rst]: I would keep it even in the actual code. Lambdas are already hard to parse as is...

**Comment by robertodr** [comment]
> Looks like the `@param` gets a colon in the docs regardless

Good to know

**Comment by stigrj** [comment]
Fixed all issues
